<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Frequently asked questions</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Frequently asked questions</h1>



<div id="how-to-be-sure-that-i-can-establish-a-connection-to-the-gwas-catalog-server" class="section level2">
<h2>1 | How to be sure that I can establish a connection to the GWAS
Catalog server?</h2>
<p>You can check that gwasrapidd is able to connect to <a href="https://www.ebi.ac.uk">https://www.ebi.ac.uk</a> by making a
connection attempt with the function
<code>is_ebi_reachable()</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">is_ebi_reachable</span>()</span></code></pre></div>
<p>Returns <code>TRUE</code> if the connection is possible, or
<code>FALSE</code> otherwise. If the connection is not possible, use the
parameter <code>chatty = TRUE</code> to learn at what point the
connection is failing.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">is_ebi_reachable</span>(<span class="at">chatty =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="what-resources-is-the-gwas-catalog-database-currently-mapped-against" class="section level2">
<h2>2 | What resources is the GWAS Catalog database currently mapped
against?</h2>
<p>The GWAS Catalog is mapped against <a href="https://www.ensembl.org">Ensembl</a>, <a href="https://www.ncbi.nlm.nih.gov/snp/">dbSNP</a> and a specific <a href="https://www.ncbi.nlm.nih.gov/grc/human">assembly version</a> of
the human genome. You can get this info with
<code>get_metadata()</code>:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_metadata</span>()</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $ensembl_release_number</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 107</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $genome_build_version</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;GRCh38.p13&quot;</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $dbsnp_version</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 154</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $usage_start_date</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2022-07-12 13:00:04 UTC&quot;</span></span></code></pre></div>
</div>
<div id="how-to-perform-batch-search-with-gwasrapidd" class="section level2">
<h2>3 | How to perform batch search with gwasrapidd?</h2>
<p>The four main retrieval functions <code>get_studies()</code>,
<code>get_associations()</code>, <code>get_variants()</code>, and
<code>get_traits()</code> allow to search by multiple values for the
same search criterion. You only need to pass a vector of queries to each
search criterion parameter. Here are some simple examples.</p>
<p>Get studies by study identifiers (GCST002420 or GCST000392):</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_studies</span>(<span class="at">study_id =</span> <span class="fu">c</span>(<span class="st">&#39;GCST002420&#39;</span>, <span class="st">&#39;GCST000392&#39;</span>))</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;studies&quot;</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;studies&quot;:</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 2 × 13</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   study_id   reported_trait          initi…¹ repli…² gxe   gxg   snp_c…³ quali…⁴ imputed pooled study…⁵ full_…⁶ user_…⁷</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                   &lt;chr&gt;   &lt;chr&gt;   &lt;lgl&gt; &lt;lgl&gt;   &lt;int&gt; &lt;chr&gt;   &lt;lgl&gt;   &lt;lgl&gt;  &lt;chr&gt;   &lt;lgl&gt;   &lt;lgl&gt;  </span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 GCST002420 Binge eating behaviour… 206 Eu… 70 Eur… FALSE FALSE 8466825 &lt;NA&gt;    TRUE    FALSE  &lt;NA&gt;    FALSE   FALSE  </span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 GCST000392 Type 1 diabetes         7,514 … 4,267 … FALSE FALSE  841622 &lt;NA&gt;    TRUE    FALSE  &lt;NA&gt;    TRUE    FALSE  </span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with abbreviated variable names ¹​initial_sample_size, ²​replication_sample_size, ³​snp_count, ⁴​qualifier,</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ⁵​study_design_comment, ⁶​full_pvalue_set, ⁷​user_requested</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;genotyping_techs&quot;:</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 2 × 2</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   study_id   genotyping_technology       </span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                       </span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 GCST002420 Genome-wide genotyping array</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 GCST000392 Genome-wide genotyping array</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;platforms&quot;:</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 2</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   study_id   manufacturer</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;       </span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 GCST002420 Affymetrix  </span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 GCST000392 Illumina    </span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 GCST000392 Affymetrix  </span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;ancestries&quot;:</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 4 × 4</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   study_id   ancestry_id type        number_of_individuals</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;                       &lt;int&gt;</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 GCST002420           1 initial                       929</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 GCST002420           2 replication                   828</span></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 GCST000392           1 initial                     16559</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 GCST000392           2 replication                 13279</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;ancestral_groups&quot;:</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 4 × 3</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   study_id   ancestry_id ancestral_group</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;          </span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 GCST002420           1 European       </span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 GCST002420           2 European       </span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 GCST000392           1 European       </span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 GCST000392           2 European       </span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;countries_of_origin&quot;:</span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 2 × 5</span></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   study_id   ancestry_id country_name major_area region</span></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt; </span></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 GCST002420           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  </span></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 GCST002420           2 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  </span></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;countries_of_recruitment&quot;:</span></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 5 × 5</span></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   study_id   ancestry_id country_name major_area       region         </span></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;            &lt;chr&gt;          </span></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 GCST002420           1 U.S.         Northern America &lt;NA&gt;           </span></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 GCST002420           2 U.S.         Northern America &lt;NA&gt;           </span></span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 GCST000392           1 U.K.         Europe           Northern Europe</span></span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 GCST000392           2 U.K.         Europe           Northern Europe</span></span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 GCST000392           2 Denmark      Europe           Northern Europe</span></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;publications&quot;:</span></span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 2 × 7</span></span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   study_id   pubmed_id publication_date publication     title                                           autho…¹ autho…²</span></span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;          &lt;int&gt; &lt;date&gt;           &lt;chr&gt;           &lt;chr&gt;                                           &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 GCST002420  24882193 2014-04-19       J Affect Disord Bipolar disorder with comorbid binge eating hi… Winham… 0000-0…</span></span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 GCST000392  19430480 2009-05-10       Nat Genet       Genome-wide association study and meta-analysi… Barret… &lt;NA&gt;   </span></span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with abbreviated variable names ¹​author_fullname, ²​author_orcid</span></span></code></pre></div>
<p>Get associations by variant identifiers (rs3798440 or rs7329174):</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_associations</span>(<span class="at">variant_id =</span> <span class="fu">c</span>(<span class="st">&#39;rs3798440&#39;</span>, <span class="st">&#39;rs7329174&#39;</span>))</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;associations&quot;</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;associations&quot;:</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 × 17</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   associ…¹ pvalue pvalu…² pvalu…³ pvalu…⁴ multi…⁵ snp_i…⁶ snp_t…⁷ stand…⁸ range or_pe…⁹ beta_…˟ beta_…˟ beta_…˟ beta_…˟</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;     &lt;int&gt;   &lt;int&gt; &lt;lgl&gt;   &lt;lgl&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 24299710  3e-10 &lt;NA&gt;          3     -10 FALSE   TRUE    novel        NA &lt;NA&gt;    NA         NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 16617     1e- 8 &lt;NA&gt;          1      -8 FALSE   FALSE   novel        NA [1.1…    1.26      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 26451     8e- 9 &lt;NA&gt;          8      -9 FALSE   FALSE   novel        NA [1.1…    1.27      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 26394     6e- 6 &lt;NA&gt;          6      -6 FALSE   FALSE   known        NA &lt;NA&gt;     1.45      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 92481688  3e- 6 &lt;NA&gt;          3      -6 FALSE   FALSE   known        NA &lt;NA&gt;    NA         NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 17433639  3e- 6 (Chine…       3      -6 FALSE   FALSE   known        NA &lt;NA&gt;     1.27      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 2 more variables: last_mapping_date &lt;dttm&gt;, last_update_date &lt;dttm&gt;, and abbreviated variable names</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ¹​association_id, ²​pvalue_description, ³​pvalue_mantissa, ⁴​pvalue_exponent, ⁵​multiple_snp_haplotype,</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ⁶​snp_interaction, ⁷​snp_type, ⁸​standard_error, ⁹​or_per_copy_number, ˟​beta_number, ˟​beta_unit, ˟​beta_direction,</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ˟​beta_description</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;loci&quot;:</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 7 × 4</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   association_id locus_id haplotype_snp_count description          </span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;             &lt;int&gt;               &lt;int&gt; &lt;chr&gt;                </span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 24299710              1                  NA SNP x SNP interaction</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 24299710              2                  NA SNP x SNP interaction</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 16617                 1                  NA Single variant       </span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 26451                 1                  NA Single variant       </span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 26394                 1                  NA Single variant       </span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 92481688              1                  NA Single variant       </span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7 17433639              1                  NA Single variant       </span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;risk_alleles&quot;:</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 7 × 7</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   association_id locus_id variant_id risk_allele risk_frequency genome_wide limited_list</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;                &lt;dbl&gt; &lt;lgl&gt;       &lt;lgl&gt;       </span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 24299710              1 rs3798440  A                   NA     TRUE        FALSE       </span></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 24299710              2 rs9350602  C                   NA     TRUE        FALSE       </span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 16617                 1 rs7329174  G                   NA     NA          NA          </span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 26451                 1 rs7329174  G                   NA     NA          NA          </span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 26394                 1 rs7329174  G                   NA     NA          NA          </span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 92481688              1 rs7329174  G                   NA     FALSE       FALSE       </span></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7 17433639              1 rs7329174  &lt;NA&gt;                 0.211 FALSE       FALSE       </span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;genes&quot;:</span></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 10 × 3</span></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    association_id locus_id gene_name   </span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;       </span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 24299710              1 MYO6        </span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 24299710              2 MYO6        </span></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 16617                 1 ELF1        </span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 26451                 1 WBP4        </span></span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 26451                 1 ELF1        </span></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 26451                 1 microRNA2276</span></span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 26451                 1 SLC25A15    </span></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 26394                 1 ELF1        </span></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 92481688              1 &lt;NA&gt;        </span></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 17433639              1 ELF1        </span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;ensembl_ids&quot;:</span></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 10 × 4</span></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    association_id locus_id gene_name    ensembl_id     </span></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;          </span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 24299710              1 MYO6         ENSG00000196586</span></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 24299710              2 MYO6         ENSG00000196586</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 16617                 1 ELF1         ENSG00000120690</span></span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 26451                 1 WBP4         ENSG00000120688</span></span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 26451                 1 ELF1         ENSG00000120690</span></span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 26451                 1 microRNA2276 &lt;NA&gt;           </span></span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 26451                 1 SLC25A15     ENSG00000102743</span></span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 26394                 1 ELF1         ENSG00000120690</span></span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 92481688              1 &lt;NA&gt;         &lt;NA&gt;           </span></span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 17433639              1 ELF1         ENSG00000120690</span></span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;entrez_ids&quot;:</span></span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 10 × 4</span></span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    association_id locus_id gene_name    entrez_id</span></span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;    </span></span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 24299710              1 MYO6         4646     </span></span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 24299710              2 MYO6         4646     </span></span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 16617                 1 ELF1         1997     </span></span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 26451                 1 WBP4         11193    </span></span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 26451                 1 ELF1         1997     </span></span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 26451                 1 microRNA2276 &lt;NA&gt;     </span></span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 26451                 1 SLC25A15     10166    </span></span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 26394                 1 ELF1         1997     </span></span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 92481688              1 &lt;NA&gt;         &lt;NA&gt;     </span></span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 17433639              1 ELF1         1997</span></span></code></pre></div>
<p>Get associations by traits (braces or binge eating or gambling):</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_associations</span>(<span class="at">efo_trait =</span> <span class="fu">c</span>(<span class="st">&#39;braces&#39;</span>, <span class="st">&#39;binge eating&#39;</span>, <span class="st">&#39;gambling&#39;</span>))</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;associations&quot;</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;associations&quot;:</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 14 × 17</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    assoc…¹ pvalue pvalu…² pvalu…³ pvalu…⁴ multi…⁵ snp_i…⁶ snp_t…⁷ stand…⁸ range or_pe…⁹ beta_…˟ beta_…˟ beta_…˟ beta_…˟</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;     &lt;int&gt;   &lt;int&gt; &lt;lgl&gt;   &lt;lgl&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 15608     4e-7 (brace…       4      -7 FALSE   FALSE   novel        NA &lt;NA&gt;    NA         NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 44592     9e-7 &lt;NA&gt;          9      -7 FALSE   FALSE   novel        NA &lt;NA&gt;     1.99      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 44589     1e-6 &lt;NA&gt;          1      -6 FALSE   FALSE   novel        NA &lt;NA&gt;     1.92      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 44590     4e-6 &lt;NA&gt;          4      -6 FALSE   FALSE   novel        NA &lt;NA&gt;     4.89      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 274608…   1e-6 &lt;NA&gt;          1      -6 FALSE   FALSE   novel        NA &lt;NA&gt;     3.23      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 274608…   1e-7 &lt;NA&gt;          1      -7 FALSE   FALSE   novel        NA &lt;NA&gt;     1.56      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 274608…   7e-7 &lt;NA&gt;          7      -7 FALSE   FALSE   novel        NA &lt;NA&gt;     1.6       NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 274608…   3e-8 &lt;NA&gt;          3      -8 FALSE   FALSE   novel        NA &lt;NA&gt;     1.92      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 274608…   1e-6 &lt;NA&gt;          1      -6 FALSE   FALSE   novel        NA &lt;NA&gt;     2.33      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 274608…   1e-8 &lt;NA&gt;          1      -8 FALSE   FALSE   novel        NA &lt;NA&gt;     2.86      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11 274608…   3e-7 &lt;NA&gt;          3      -7 FALSE   FALSE   novel        NA &lt;NA&gt;     1.75      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12 274608…   3e-7 &lt;NA&gt;          3      -7 FALSE   FALSE   novel        NA &lt;NA&gt;     2.08      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13 274608…   1e-6 &lt;NA&gt;          1      -6 FALSE   FALSE   novel        NA &lt;NA&gt;     2         NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 274608…   9e-8 &lt;NA&gt;          9      -8 FALSE   FALSE   novel        NA &lt;NA&gt;     3.85      NA &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   </span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 2 more variables: last_mapping_date &lt;dttm&gt;, last_update_date &lt;dttm&gt;, and abbreviated variable names</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ¹​association_id, ²​pvalue_description, ³​pvalue_mantissa, ⁴​pvalue_exponent, ⁵​multiple_snp_haplotype,</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ⁶​snp_interaction, ⁷​snp_type, ⁸​standard_error, ⁹​or_per_copy_number, ˟​beta_number, ˟​beta_unit, ˟​beta_direction,</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   ˟​beta_description</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;loci&quot;:</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 14 × 4</span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    association_id locus_id haplotype_snp_count description   </span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;             &lt;int&gt;               &lt;int&gt; &lt;chr&gt;         </span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 15608                 1                  NA Single variant</span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 44592                 1                  NA Single variant</span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 44589                 1                  NA Single variant</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 44590                 1                  NA Single variant</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 27460823              1                  NA Single variant</span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 27460811              1                  NA Single variant</span></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 27460817              1                  NA Single variant</span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 27460805              1                  NA Single variant</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 27460830              1                  NA Single variant</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 27460844              1                  NA Single variant</span></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11 27460858              1                  NA Single variant</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12 27460864              1                  NA Single variant</span></span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13 27460870              1                  NA Single variant</span></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 27460851              1                  NA Single variant</span></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;risk_alleles&quot;:</span></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 14 × 7</span></span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    association_id locus_id variant_id  risk_allele risk_frequency genome_wide limited_list</span></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;                &lt;dbl&gt; &lt;lgl&gt;       &lt;lgl&gt;       </span></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 15608                 1 rs1535480   &lt;NA&gt;                 NA    NA          NA          </span></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 44592                 1 rs6006893   &lt;NA&gt;                 NA    NA          NA          </span></span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 44589                 1 rs10198175  &lt;NA&gt;                 NA    NA          NA          </span></span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 44590                 1 rs13233490  &lt;NA&gt;                 NA    NA          NA          </span></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 27460823              1 rs182107583 C                     0.04 FALSE       FALSE       </span></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 27460811              1 rs7904579   G                     0.37 FALSE       FALSE       </span></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 27460817              1 rs1950038   T                     0.3  FALSE       FALSE       </span></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 27460805              1 rs726170    T                     0.12 FALSE       FALSE       </span></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 27460830              1 rs76087671  T                     0.05 FALSE       FALSE       </span></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 27460844              1 rs111940429 T                     0.04 FALSE       FALSE       </span></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11 27460858              1 rs7337127   T                     0.15 FALSE       FALSE       </span></span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12 27460864              1 rs145763646 A                     0.1  FALSE       FALSE       </span></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13 27460870              1 rs73057489  C                     0.07 FALSE       FALSE       </span></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 27460851              1 rs17810023  T                     0.02 FALSE       FALSE       </span></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;genes&quot;:</span></span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 15 × 3</span></span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    association_id locus_id gene_name   </span></span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;       </span></span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 15608                 1 &lt;NA&gt;        </span></span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 44592                 1 PRR5        </span></span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 44589                 1 APOB        </span></span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 44590                 1 PER4        </span></span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 27460823              1 LOC101929321</span></span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 27460811              1 CUBN        </span></span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 27460817              1 Intergenic  </span></span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 27460805              1 PRR5        </span></span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 27460805              1 ARHGAP8     </span></span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 27460830              1 intergenic  </span></span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11 27460844              1 AC096669.1  </span></span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12 27460858              1 intergenic  </span></span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13 27460864              1 SLC25A26    </span></span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 27460870              1 intergenic  </span></span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15 27460851              1 RP11-250B2.3</span></span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;ensembl_ids&quot;:</span></span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 15 × 4</span></span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    association_id locus_id gene_name    ensembl_id     </span></span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;          </span></span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 15608                 1 &lt;NA&gt;         &lt;NA&gt;           </span></span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 44592                 1 PRR5         ENSG00000186654</span></span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 44589                 1 APOB         ENSG00000084674</span></span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 44590                 1 PER4         &lt;NA&gt;           </span></span>
<span id="cb6-92"><a href="#cb6-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 27460823              1 LOC101929321 &lt;NA&gt;           </span></span>
<span id="cb6-93"><a href="#cb6-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 27460811              1 CUBN         ENSG00000107611</span></span>
<span id="cb6-94"><a href="#cb6-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 27460817              1 Intergenic   &lt;NA&gt;           </span></span>
<span id="cb6-95"><a href="#cb6-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 27460805              1 PRR5         ENSG00000186654</span></span>
<span id="cb6-96"><a href="#cb6-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 27460805              1 ARHGAP8      ENSG00000241484</span></span>
<span id="cb6-97"><a href="#cb6-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 27460830              1 intergenic   &lt;NA&gt;           </span></span>
<span id="cb6-98"><a href="#cb6-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11 27460844              1 AC096669.1   ENSG00000225588</span></span>
<span id="cb6-99"><a href="#cb6-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12 27460858              1 intergenic   &lt;NA&gt;           </span></span>
<span id="cb6-100"><a href="#cb6-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13 27460864              1 SLC25A26     ENSG00000144741</span></span>
<span id="cb6-101"><a href="#cb6-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 27460870              1 intergenic   &lt;NA&gt;           </span></span>
<span id="cb6-102"><a href="#cb6-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15 27460851              1 RP11-250B2.3 &lt;NA&gt;           </span></span>
<span id="cb6-103"><a href="#cb6-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-104"><a href="#cb6-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;entrez_ids&quot;:</span></span>
<span id="cb6-105"><a href="#cb6-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 15 × 4</span></span>
<span id="cb6-106"><a href="#cb6-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    association_id locus_id gene_name    entrez_id</span></span>
<span id="cb6-107"><a href="#cb6-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;    </span></span>
<span id="cb6-108"><a href="#cb6-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 15608                 1 &lt;NA&gt;         &lt;NA&gt;     </span></span>
<span id="cb6-109"><a href="#cb6-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 44592                 1 PRR5         55615    </span></span>
<span id="cb6-110"><a href="#cb6-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 44589                 1 APOB         338      </span></span>
<span id="cb6-111"><a href="#cb6-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 44590                 1 PER4         &lt;NA&gt;     </span></span>
<span id="cb6-112"><a href="#cb6-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 27460823              1 LOC101929321 101929321</span></span>
<span id="cb6-113"><a href="#cb6-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 27460811              1 CUBN         8029     </span></span>
<span id="cb6-114"><a href="#cb6-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 27460817              1 Intergenic   &lt;NA&gt;     </span></span>
<span id="cb6-115"><a href="#cb6-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 27460805              1 PRR5         55615    </span></span>
<span id="cb6-116"><a href="#cb6-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 27460805              1 ARHGAP8      23779    </span></span>
<span id="cb6-117"><a href="#cb6-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 27460830              1 intergenic   &lt;NA&gt;     </span></span>
<span id="cb6-118"><a href="#cb6-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11 27460844              1 AC096669.1   &lt;NA&gt;     </span></span>
<span id="cb6-119"><a href="#cb6-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12 27460858              1 intergenic   &lt;NA&gt;     </span></span>
<span id="cb6-120"><a href="#cb6-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13 27460864              1 SLC25A26     115286   </span></span>
<span id="cb6-121"><a href="#cb6-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 27460870              1 intergenic   &lt;NA&gt;     </span></span>
<span id="cb6-122"><a href="#cb6-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15 27460851              1 RP11-250B2.3 &lt;NA&gt;</span></span></code></pre></div>
<p>Get traits by PubMed identifiers (24882193 or 22780124):</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_traits</span>(<span class="at">pubmed_id =</span> <span class="fu">c</span>(<span class="st">&#39;24882193&#39;</span>, <span class="st">&#39;22780124&#39;</span>))</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;traits&quot;</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Slot &quot;traits&quot;:</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 3</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   efo_id      trait              uri                                 </span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;              &lt;chr&gt;                               </span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 EFO_0005924 binge eating       http://www.ebi.ac.uk/efo/EFO_0005924</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 EFO_0000289 bipolar disorder   http://www.ebi.ac.uk/efo/EFO_0000289</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 EFO_0004699 gambling behaviour http://www.ebi.ac.uk/efo/EFO_0004699</span></span></code></pre></div>
<p>The only search parameters that are not vectorised are
<code>user_requested</code> and <code>full_pvalue_set</code> from
<code>get_studies()</code>. These parameters are not vectorised because
they take boolean values (<code>TRUE</code> or <code>FALSE</code>) and
thus only one of the values is sensical to be used as a query at a given
time.</p>
</div>
<div id="what-is-the-difference-between-a-trait-and-a-reported-trait" class="section level2">
<h2>4 | What is the difference between a trait and a reported
trait?</h2>
<p>There are two levels of trait description in the GWAS Catalog: (EFO)
trait and reported trait.</p>
<p>Studies are assigned one or more terms from the <a href="https://www.ebi.ac.uk/efo/">Experimental Factor Ontology
(EFO)</a>, i.e., an EFO trait, or simply trait, that best represents the
phenotype under investigation.</p>
<p>In addition, each study is also assigned a free text reported trait.
This is written by the GWAS Catalog curators and reflects the author
language, and where necessary, it includes more specific and detailed
description of the experimental design, e.g., interaction studies or
studies with a background trait.</p>
<p>As an example take the study with accession identifier
<code>GCST000206</code> by EM Behrens et al. (2008). We can get the EFO
trait with <code>get_traits()</code> and the reported trait with
<code>get_studies()</code>:</p>
<p>The (EFO) trait for the Behrens study is
<code>chronic childhood arthritis</code>:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>efo_trait <span class="ot">&lt;-</span> <span class="fu">get_traits</span>(<span class="at">study_id =</span> <span class="st">&#39;GCST000206&#39;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>efo_trait<span class="sc">@</span>traits<span class="sc">$</span>trait</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;juvenile idiopathic arthritis&quot;</span></span></code></pre></div>
<p>whereas the reported trait is
<code>Arthritis (juvenile idiopathic)</code>:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>study <span class="ot">&lt;-</span> <span class="fu">get_studies</span>(<span class="at">study_id =</span> <span class="st">&#39;GCST000206&#39;</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>study<span class="sc">@</span>studies<span class="sc">$</span>reported_trait</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Arthritis (juvenile idiopathic)&quot;</span></span></code></pre></div>
</div>
<div id="genomic-coordinates-of-genomic-contexts-seem-to-be-wrong" class="section level2">
<h2>5 | Genomic coordinates of genomic contexts seem to be wrong?</h2>
<p>The REST API response for variants contains an element named
<em>genomic contexts</em>. This element is mapped onto the table
<code>genomic_contexts</code> of a <a href="https://rmagno.eu/gwasrapidd/reference/variants-class.html">variants</a>
S4 object.</p>
<p>Now, there is indeed a server-side bug with the column
<code>chromosome_position</code> of the <code>genomic_contexts</code>
table: the chromosome position returned is that of the variant and not
of the gene (<em>genomic context</em>) as it should be.</p>
<p>The GWAS Catalog team is aware of this bug, and they plan to fix it,
eventually. For the time being, just do not rely on
<code>chromosome_position</code> of the <code>genomic_contexts</code>
table.</p>
</div>
<div id="how-to-search-for-variants-within-a-certain-genomic-region" class="section level2">
<h2>6 | How to search for variants within a certain genomic region?</h2>
<div id="single-genomic-range" class="section level3">
<h3>Single genomic range</h3>
<p>For this you may use the function <code>get_variants()</code> with
parameter <code>genomic_range</code>.</p>
<p>For example, to search for variants in chromosome Y in the interval
14692000–14695000, you start by defining a list of 3 elements:
<code>chromosome</code>, <code>start</code> and <code>end</code> that
specify your genomic range:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># &#39;chromosome&#39; names are case sensitive, and should be uppercase.</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># &#39;start&#39; and &#39;end&#39; positions should be integer vectors.</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>my_genomic_range <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">chromosome =</span> <span class="st">&#39;Y&#39;</span>,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">start =</span> 14692000L,</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">end =</span> 14695000L)</span></code></pre></div>
<p>Now you can use <code>my_genomic_range</code> to retrieve the
variants:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>chr_Y_variants <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(<span class="at">genomic_range =</span> my_genomic_range)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>chr_Y_variants<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>, <span class="st">&#39;functional_class&#39;</span>)]</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 2</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   variant_id functional_class</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;           </span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 rs2115848  intron_variant</span></span></code></pre></div>
</div>
<div id="multiple-genomic-ranges" class="section level3">
<h3>Multiple genomic ranges</h3>
<p>To search in multiple regions, construct your genomic range list with
those locations just like in the previous example. For example, let’s
search now for variants in chromosome X and Y, both in range
13000000–15000000:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>my_genomic_range <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">chromosome =</span> <span class="fu">c</span>(<span class="st">&#39;X&#39;</span>, <span class="st">&#39;Y&#39;</span>),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">start =</span> <span class="fu">c</span>(13000000L, 13000000L),</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">end =</span> <span class="fu">c</span>(15000000L, 15000000L))</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>chr_XY_variants <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(<span class="at">genomic_range =</span> my_genomic_range)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>chr_XY_variants<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&#39;chromosome_position&#39;</span>)]</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 24 × 3</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    variant_id  chromosome_name chromosome_position</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;                         &lt;int&gt;</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 rs111689944 X                          13494655</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 rs7063195   X                          13492087</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 rs5978649   X                          13596999</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 rs5934106   X                          13488259</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 rs749624882 X                          14929148</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 rs61273829  X                          14150876</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 rs6528024   X                          13912803</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 rs66819623  X                          13936278</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 rs138331350 X                          14841981</span></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 rs5980075   X                          14932409</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 14 more rows</span></span></code></pre></div>
</div>
<div id="searching-variants-by-cytogenetic-regions" class="section level3">
<h3>Searching variants by cytogenetic regions</h3>
<p>To search for variants within a cytogenetic band you can use the
parameter <code>cytogenetic_band</code> of <code>get_variants()</code>.
Here is an example, again for chromosome Y, using the cytogenetic band
<code>&#39;Yq11.221&#39;</code> as query:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>my_variants <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(<span class="at">cytogenetic_band =</span> <span class="st">&#39;Yq11.221&#39;</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>my_variants<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_position&#39;</span>)]</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 3</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   variant_id chromosome_name chromosome_position</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                         &lt;int&gt;</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 rs2032624  Y                          12914512</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 rs2115848  Y                          14692972</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 rs2032658  Y                          13470103</span></span></code></pre></div>
<p>How to know what are the cytogenetic bands for querying? We provide a
dataset (dataframe) named <code>cytogenetic_bands</code> that you can
use:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ?cytogenetic_bands for more details.</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>cytogenetic_bands</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 862 × 8</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    cytogenetic_band chromosome    start      end  length assembly stain  last_download_date      </span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt;         &lt;int&gt;    &lt;int&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;  &lt;chr&gt;                   </span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 1p36.33          1                 1  2300000 2300000 GRCh38   gneg   Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 1p36.32          1           2300001  5300000 3000000 GRCh38   gpos25 Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 1p36.31          1           5300001  7100000 1800000 GRCh38   gneg   Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 1p36.23          1           7100001  9100000 2000000 GRCh38   gpos25 Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 1p36.22          1           9100001 12500000 3400000 GRCh38   gneg   Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 1p36.21          1          12500001 15900000 3400000 GRCh38   gpos50 Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 1p36.13          1          15900001 20100000 4200000 GRCh38   gneg   Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 1p36.12          1          20100001 23600000 3500000 GRCh38   gpos25 Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 1p36.11          1          23600001 27600000 4000000 GRCh38   gneg   Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 1p35.3           1          27600001 29900000 2300000 GRCh38   gpos25 Thu Jun 27 15:09:22 2019</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 852 more rows</span></span></code></pre></div>
<p>Let’s say you want to search for all variants in the shorter arm (p)
of chromosome 21, you can take advantage of the
<code>cytogenetic_bands</code> to get all the corresponding cytogenetic
band names:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Install package dplyr if you do not have it.</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>chr21_p_bands <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(cytogenetic_bands, <span class="fu">grepl</span>(<span class="st">&#39;^21p&#39;</span>, cytogenetic_band)) <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(cytogenetic_band)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>chr21_p_bands</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;21p13&quot;   &quot;21p12&quot;   &quot;21p11.2&quot; &quot;21p11.1&quot;</span></span></code></pre></div>
<p>Now search by <code>cytogenetic_band</code>:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>my_variants <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(<span class="at">cytogenetic_band =</span> chr21_p_bands)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>my_variants<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_position&#39;</span>,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_region&#39;</span>)]</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 4 × 4</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   variant_id  chromosome_name chromosome_position chromosome_region</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;            </span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 rs552964654 21                          9053097 21p11.2          </span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 rs10439884  21                         10540506 21p11.2          </span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 rs75275574  21                          9040338 21p11.2          </span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 rs240444    21                         10510446 21p11.2</span></span></code></pre></div>
</div>
</div>
<div id="genomic-range-for-an-entire-chromosome" class="section level2">
<h2>7 | Genomic range for an entire chromosome?</h2>
<p>You can get the total length of a chromosome by using the provided
data set: <code>cytogenetic_bands</code>. Here is an example for
chromosome 15:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Install dplyr first.</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(cytogenetic_bands, chromosome <span class="sc">==</span> <span class="st">&#39;15&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">chromosome =</span> dplyr<span class="sc">::</span><span class="fu">first</span>(chromosome),</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">start =</span> <span class="fu">min</span>(start),</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">end =</span> <span class="fu">max</span>(end)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                   )</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 3</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   chromosome start       end</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;int&gt;     &lt;int&gt;</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 15             1 101991189</span></span></code></pre></div>
</div>
<div id="how-to-keep-track-of-which-queries-generated-which-results" class="section level2">
<h2>8 | How to keep track of which queries generated which results?</h2>
<p>Currently, there is not an implemented solution in
<code>{gwasrapidd}</code>. For example, if you search for variants by
EFO identifier (<code>efo_id</code>):</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>my_efo_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;EFO_0005543&#39;</span>, <span class="st">&#39;EFO_0004762&#39;</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>my_variants <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(<span class="at">efo_id =</span> my_efo_ids)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>my_variants<span class="sc">@</span>variants<span class="sc">$</span>variant_id</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [1] &quot;rs75061358&quot;  &quot;rs4252707&quot;   &quot;rs72714270&quot;  &quot;rs1106639&quot;   &quot;rs2736100&quot;   &quot;rs648044&quot;    &quot;rs78378222&quot;  &quot;rs7572263&quot;  </span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [9] &quot;rs498872&quot;    &quot;rs10852606&quot;  &quot;rs4975538&quot;   &quot;rs12803321&quot;  &quot;rs10131032&quot;  &quot;rs6010620&quot;   &quot;rs4774756&quot;   &quot;rs1938964&quot;  </span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [17] &quot;rs10927065&quot;  &quot;rs5839764&quot;   &quot;rs7107785&quot;   &quot;rs634537&quot;    &quot;rs72714236&quot;  &quot;rs12076373&quot;  &quot;rs1275600&quot;   &quot;rs11706832&quot; </span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [25] &quot;rs111976262&quot; &quot;rs9841110&quot;   &quot;rs80351950&quot;  &quot;rs11598018&quot;  &quot;rs2297440&quot;   &quot;rs77633900&quot;  &quot;rs3751667&quot;   &quot;rs11599775&quot; </span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [33] &quot;rs10842893&quot;  &quot;rs320337&quot;    &quot;rs12752552&quot;  &quot;rs7125115&quot;   &quot;rs688755&quot;    &quot;rs2297433&quot;   &quot;rs10069690&quot;  &quot;rs11979158&quot; </span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [41] &quot;rs2235573&quot;   &quot;rs3772190&quot;   &quot;rs4977756&quot;   &quot;rs55705857&quot;  &quot;rs11233250&quot;  &quot;rs78355601&quot;  &quot;rs7043199&quot;   &quot;rs187918360&quot;</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [49] &quot;rs75455100&quot;  &quot;rs10886366&quot;  &quot;rs6598475&quot;   &quot;rs150821445&quot; &quot;rs6479877&quot;   &quot;rs73418461&quot;  &quot;rs9787438&quot;   &quot;rs6993770&quot;  </span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [57] &quot;rs4782371&quot;   &quot;rs2375981&quot;   &quot;rs9472155&quot;   &quot;rs34528081&quot;  &quot;rs144160960&quot; &quot;rs12445232&quot;  &quot;rs10934631&quot;  &quot;rs71779653&quot; </span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [65] &quot;rs117580153&quot; &quot;rs12456390&quot;  &quot;rs10822155&quot;  &quot;rs7767396&quot;   &quot;rs59706856&quot;  &quot;rs73872715&quot;  &quot;rs181216459&quot; &quot;rs10761731&quot; </span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [73] &quot;rs10738760&quot;  &quot;rs186066666&quot; &quot;rs8045833&quot;   &quot;rs11392719&quot;  &quot;rs180936035&quot; &quot;rs4513773&quot;   &quot;rs118086804&quot; &quot;rs114694170&quot;</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [81] &quot;rs181558074&quot; &quot;rs7763358&quot;   &quot;rs6722871&quot;   &quot;rs2304058&quot;   &quot;rs6921438&quot;   &quot;rs61829244&quot;  &quot;rs191332118&quot; &quot;rs12214617&quot; </span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [89] &quot;rs1740073&quot;   &quot;rs139893147&quot; &quot;rs77961527&quot;  &quot;rs7030781&quot;   &quot;rs61818787&quot;  &quot;rs550057&quot;    &quot;rs11639051&quot;  &quot;rs10411345&quot; </span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [97] &quot;rs10761750&quot;  &quot;rs4714729&quot;   &quot;rs4082730&quot;   &quot;rs9332599&quot;   &quot;rs10761741&quot;  &quot;rs12199215&quot;  &quot;rs186725382&quot; &quot;rs144820908&quot;</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [105] &quot;rs10153304&quot;  &quot;rs112215592&quot; &quot;rs13206436&quot;  &quot;rs2639990&quot;   &quot;rs6475938&quot;   &quot;rs144085478&quot; &quot;rs34524635&quot;  &quot;rs55864163&quot; </span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [113] &quot;rs143479231&quot; &quot;rs34881325&quot;</span></span></code></pre></div>
<p>So it is not immediately obvious which variants resulted from the
query <code>&#39;EFO_0005543&#39;</code> or <code>&#39;EFO_0004762&#39;</code>.</p>
<p>A possible workaround is to make multiple independent queries and
save your results in a list whose names are the respective queries:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Install purrr first.</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Add names to my_efo_ids</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(my_efo_ids) <span class="ot">&lt;-</span> my_efo_ids</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>my_variants <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(my_efo_ids, <span class="sc">~</span> <span class="fu">get_variants</span>(<span class="at">efo_id =</span> .x))</span></code></pre></div>
<p>Now you can see which variants are associated with each EFO
identifier.</p>
<p>For <code>&#39;EFO_0005543&#39;</code> we got the following variants:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>my_variants[[<span class="st">&#39;EFO_0005543&#39;</span>]]<span class="sc">@</span>variants<span class="sc">$</span>variant_id</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;rs75061358&quot;  &quot;rs4252707&quot;   &quot;rs72714270&quot;  &quot;rs1106639&quot;   &quot;rs2736100&quot;   &quot;rs648044&quot;    &quot;rs78378222&quot;  &quot;rs7572263&quot;  </span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;rs498872&quot;    &quot;rs10852606&quot;  &quot;rs4975538&quot;   &quot;rs12803321&quot;  &quot;rs10131032&quot;  &quot;rs6010620&quot;   &quot;rs4774756&quot;   &quot;rs1938964&quot;  </span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] &quot;rs10927065&quot;  &quot;rs5839764&quot;   &quot;rs7107785&quot;   &quot;rs634537&quot;    &quot;rs72714236&quot;  &quot;rs12076373&quot;  &quot;rs1275600&quot;   &quot;rs11706832&quot; </span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] &quot;rs111976262&quot; &quot;rs9841110&quot;   &quot;rs80351950&quot;  &quot;rs11598018&quot;  &quot;rs2297440&quot;   &quot;rs77633900&quot;  &quot;rs3751667&quot;   &quot;rs11599775&quot; </span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [33] &quot;rs10842893&quot;  &quot;rs320337&quot;    &quot;rs12752552&quot;  &quot;rs7125115&quot;   &quot;rs688755&quot;    &quot;rs2297433&quot;   &quot;rs10069690&quot;  &quot;rs11979158&quot; </span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [41] &quot;rs2235573&quot;   &quot;rs3772190&quot;   &quot;rs4977756&quot;   &quot;rs55705857&quot;  &quot;rs11233250&quot;</span></span></code></pre></div>
<p>And for <code>&#39;EFO_0004762&#39;</code>:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>my_variants[[<span class="st">&#39;EFO_0004762&#39;</span>]]<span class="sc">@</span>variants<span class="sc">$</span>variant_id</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;rs78355601&quot;  &quot;rs7043199&quot;   &quot;rs187918360&quot; &quot;rs75455100&quot;  &quot;rs10886366&quot;  &quot;rs6598475&quot;   &quot;rs150821445&quot; &quot;rs6479877&quot;  </span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;rs73418461&quot;  &quot;rs9787438&quot;   &quot;rs6993770&quot;   &quot;rs4782371&quot;   &quot;rs2375981&quot;   &quot;rs9472155&quot;   &quot;rs34528081&quot;  &quot;rs144160960&quot;</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] &quot;rs12445232&quot;  &quot;rs10934631&quot;  &quot;rs71779653&quot;  &quot;rs117580153&quot; &quot;rs12456390&quot;  &quot;rs10822155&quot;  &quot;rs7767396&quot;   &quot;rs59706856&quot; </span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] &quot;rs73872715&quot;  &quot;rs181216459&quot; &quot;rs10761731&quot;  &quot;rs10738760&quot;  &quot;rs186066666&quot; &quot;rs8045833&quot;   &quot;rs11392719&quot;  &quot;rs180936035&quot;</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [33] &quot;rs4513773&quot;   &quot;rs118086804&quot; &quot;rs114694170&quot; &quot;rs181558074&quot; &quot;rs7763358&quot;   &quot;rs6722871&quot;   &quot;rs2304058&quot;   &quot;rs6921438&quot;  </span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [41] &quot;rs61829244&quot;  &quot;rs191332118&quot; &quot;rs12214617&quot;  &quot;rs1740073&quot;   &quot;rs139893147&quot; &quot;rs77961527&quot;  &quot;rs7030781&quot;   &quot;rs61818787&quot; </span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [49] &quot;rs550057&quot;    &quot;rs11639051&quot;  &quot;rs10411345&quot;  &quot;rs10761750&quot;  &quot;rs4714729&quot;   &quot;rs4082730&quot;   &quot;rs9332599&quot;   &quot;rs10761741&quot; </span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [57] &quot;rs12199215&quot;  &quot;rs186725382&quot; &quot;rs144820908&quot; &quot;rs10153304&quot;  &quot;rs112215592&quot; &quot;rs13206436&quot;  &quot;rs2639990&quot;   &quot;rs6475938&quot;  </span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [65] &quot;rs144085478&quot; &quot;rs34524635&quot;  &quot;rs55864163&quot;  &quot;rs143479231&quot; &quot;rs34881325&quot;</span></span></code></pre></div>
</div>
<div id="how-to-combine-results-from-multiple-queries" class="section level2">
<h2>9 | How to combine results from multiple queries?</h2>
<p>The four main retrieval functions <code>get_studies()</code>,
<code>get_associations()</code>, <code>get_variants()</code> and
<code>get_traits()</code> all allow you to search multiple criteria at
once. You can then combine results in an OR or AND fashion using the
parameter <code>set_operation</code>.</p>
<p>Use <code>set_operation = &#39;union&#39;</code> to combine results in an OR
fashion:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>my_variants_OR <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">efo_trait =</span> <span class="st">&#39;triple-negative breast cancer&#39;</span>,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene_name =</span> <span class="st">&#39;MDM4&#39;</span>,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">set_operation =</span> <span class="st">&#39;union&#39;</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>my_variants_OR<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>,</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_position&#39;</span>,</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_region&#39;</span>)]</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 56 × 4</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    variant_id  chromosome_name chromosome_position chromosome_region</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;            </span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 rs3747636   1                         204434531 1q32.1           </span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 rs10793765  1                         204580247 1q32.1           </span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 rs116661163 1                         204641544 1q32.1           </span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 rs12083887  1                         118339066 1p12             </span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 rs12143943  1                         204602943 1q32.1           </span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 rs1008833   1                         204457167 1q32.1           </span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 rs2137255   1                         204457245 1q32.1           </span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 rs2290854   1                         204546897 1q32.1           </span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 rs12036042  1                         204500186 1q32.1           </span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 rs3789045   1                         204617684 1q32.1           </span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 46 more rows</span></span></code></pre></div>
<p>The code above retrieves variants whose associated
<code>efo_trait</code> is equal to
<code>&#39;triple-negative breast cancer&#39;</code> or variants that are
associated with gene <code>&#39;MDM4&#39;</code>.</p>
<p>Alternatively, we may use <code>set_operation = &#39;intersection&#39;</code>
to combine results in an AND fashion:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>my_variants_AND <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">efo_trait =</span> <span class="st">&#39;triple-negative breast cancer&#39;</span>,</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene_name =</span> <span class="st">&#39;MDM4&#39;</span>,</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">set_operation =</span> <span class="st">&#39;intersection&#39;</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>my_variants_AND<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>,</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_position&#39;</span>,</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_region&#39;</span>)]</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 0 × 4</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 4 variables: variant_id &lt;chr&gt;, chromosome_name &lt;chr&gt;, chromosome_position &lt;int&gt;, chromosome_region &lt;chr&gt;</span></span></code></pre></div>
<p>With <code>set_operation = &#39;intersection&#39;</code>, as in the code
above, we get variants whose associated <code>efo_trait</code> is equal
to <code>&#39;triple-negative breast cancer&#39;</code> and that are associated
with gene <code>&#39;MDM4&#39;</code>, i.e., only variants meeting both
conditions simultaneously are retrieved.</p>
<p>Please note that almost all search criteria to be used with the
retrieval functions are vectorised, meaning that you can use multiple
values with the same search criterion. In these cases results are always
combined in an OR fashion.</p>
<p>In the following example, we will be using the gene name as the only
search criterion. If we pass a vector of gene names then we get all
variants that are associated with EITHER (OR) genes.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>my_variants <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(<span class="at">gene_name =</span> <span class="fu">c</span>(<span class="st">&#39;RNU6-367P&#39;</span>, <span class="st">&#39;ABHD5&#39;</span>))</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; downloading [=======================================================================================] 100% eta: 0s</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; downloading [=======================================================================================] 100% eta: 0s</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>my_variants<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>,</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_position&#39;</span>,</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_region&#39;</span>)]</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 33 × 4</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    variant_id  chromosome_name chromosome_position chromosome_region</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;            </span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 rs142242702 3                          44122660 3p21.32          </span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 rs1554654   3                          44002852 3p21.33          </span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 rs7619427   3                          44035549 3p21.33          </span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 rs11718455  3                          44015406 3p21.33          </span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 rs73076675  3                          43924731 3p21.33          </span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 rs113706999 3                          44117664 3p21.32          </span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 rs6441814   3                          44007622 3p21.33          </span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 rs4017425   3                          43987272 3p21.33          </span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 rs79644353  3                          44135707 3p21.32          </span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 rs7619544   3                          43852053 3p21.33          </span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 23 more rows</span></span></code></pre></div>
<p>In this case we retrieved 33 variants. Please note that the
<code>set_operation</code> parameter does not affect this result. The
<code>set_operation</code> only controls the function behaviour when
combining results from different criteria, e.g., when using
<code>efo_trait</code> and <code>gene_name</code>.</p>
<p>To retrieve variants that are concomitantly associated with genes
RNU6-367P and ABHD5, the user needs to place these queries separately
and then intersect them — using the <code>intersect()</code> function,
i.e., combining in an AND fashion. Here we start by retrieving variants
associated with gene RNU6-367P:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>my_variants1 <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(<span class="at">gene_name =</span> <span class="st">&#39;RNU6-367P&#39;</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; downloading [=======================================================================================] 100% eta: 0s</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>my_variants1<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>,</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_position&#39;</span>,</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_region&#39;</span>)]</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 19 × 4</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    variant_id  chromosome_name chromosome_position chromosome_region</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;            </span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 rs142242702 3                          44122660 3p21.32          </span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 rs1554654   3                          44002852 3p21.33          </span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 rs7619427   3                          44035549 3p21.33          </span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 rs11718455  3                          44015406 3p21.33          </span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 rs73076675  3                          43924731 3p21.33          </span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 rs113706999 3                          44117664 3p21.32          </span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 rs6441814   3                          44007622 3p21.33          </span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 rs4017425   3                          43987272 3p21.33          </span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 rs79644353  3                          44135707 3p21.32          </span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 rs7619544   3                          43852053 3p21.33          </span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11 rs58492658  3                          44098920 3p21.33          </span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12 rs7635417   3                          44032957 3p21.33          </span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13 rs6441813   3                          43999831 3p21.33          </span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 rs189710954 3                          43800315 3p21.33          </span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15 rs11130017  3                          43932687 3p21.33          </span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16 rs75619216  3                          43912969 3p21.33          </span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 17 rs1491708   3                          43870086 3p21.33          </span></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 18 rs55807460  3                          43903734 3p21.33          </span></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 19 rs6806257   3                          43878471 3p21.33</span></span></code></pre></div>
<p>There are 19 variants associated with gene RNU6-367P. Now, for gene
ABHD5:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>my_variants2 <span class="ot">&lt;-</span> <span class="fu">get_variants</span>(<span class="at">gene_name =</span> <span class="st">&#39;ABHD5&#39;</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; downloading [=========================================================&gt;-----------------------------] 67% eta: 4s</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; downloading [=======================================================================================] 100% eta: 0s</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>my_variants2<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>,</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_position&#39;</span>,</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_region&#39;</span>)]</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 23 × 4</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    variant_id  chromosome_name chromosome_position chromosome_region</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;            </span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1 rs191867523 3                          43774449 3p21.33          </span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2 rs115421670 3                          43737217 3p21.33          </span></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3 rs6772840   3                          43812828 3p21.33          </span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4 rs11720728  3                          43832420 3p21.33          </span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5 rs142404191 3                          43641812 3p21.33          </span></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6 rs192944990 3                          43826178 3p21.33          </span></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7 rs182923613 3                          43791731 3p21.33          </span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8 rs73087085  3                          43687796 3p21.33          </span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9 rs75594032  3                          43744082 3p21.33          </span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 rs1468602   3                          43701242 3p21.33          </span></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 13 more rows</span></span></code></pre></div>
<p>There are 23 variants associated with gene ABHD5. To find those
variants simultaneously associated with both genes, you can intersect
the two <a href="https://rmagno.eu/gwasrapidd/reference/variants-class.html">variants</a>
objects using <code>gwasrapidd::intersect()</code>:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>variants_intersect <span class="ot">&lt;-</span> gwasrapidd<span class="sc">::</span><span class="fu">intersect</span>(my_variants1, my_variants2)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>variants_intersect<span class="sc">@</span>variants[<span class="fu">c</span>(<span class="st">&#39;variant_id&#39;</span>,</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_position&#39;</span>,</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&#39;chromosome_region&#39;</span>)]</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 9 × 4</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   variant_id  chromosome_name chromosome_position chromosome_region</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;            </span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 rs4017425   3                          43987272 3p21.33          </span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 rs7635417   3                          44032957 3p21.33          </span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 rs6441813   3                          43999831 3p21.33          </span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 rs189710954 3                          43800315 3p21.33          </span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 rs11130017  3                          43932687 3p21.33          </span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 rs75619216  3                          43912969 3p21.33          </span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7 rs1491708   3                          43870086 3p21.33          </span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8 rs55807460  3                          43903734 3p21.33          </span></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9 rs6806257   3                          43878471 3p21.33</span></span></code></pre></div>
<p>Apparently only 9 variant(s) are related to both genes RNU6-367P and
ABHD5.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
